tests/yast2_gui/yast2_lang.pm

```diff
@@ -26,7 +26,9 @@ sub run {
 
     # check language details and change detailed locale setting
     assert_and_click 'yast2-lang_details';
-    assert_and_click 'yast2-lang_detailed-locale-setting';
+    assert_screen 'yast2-lang_detailed-locale-setting';
+    send_key 'alt-d';
+    send_key_until_needlematch('yast2-lang_detailed-locale-setting_en_GB', 'spc', 5, 2);
```

Comment:
you're assuming `alt-d` hits the VM and combobox is selected and the problems is afterwards not on the selection. So in my understanding `assert_and_click` is failing for this architecture. So instead of doing this hacky 'spc' thing, it would be easier just to avoid `assert_and_click` before and after, so you can simplify the flow with just moving up:
`send_key_until_needlematch('yast2-lang_detailed-locale-setting_en_GB', 'up')`

Reply:
The issue was that combobox is not expanded after sending `assert_and_click 'yast2-lang_detailed-locale-setting';`.

In my solution I'm just selecting the combobox with `alt-d` and then pressing `space` key to expand it. Then the steps are same as before. 

The only "hacky" here is that I do not trust that it will be expanded from the first time and using `send_key_until_needlematch` instead of regular `send_key` to press `space` again if the key was missed. 

Sorry, I didn't get how pressing 'up' will help to expand the dropdown:)

Reply:
pressing 'up' will not expand the dropdown, it will just move to the right item, simplifying the flow and avoiding totally `assert_and_click` for this case.

Reply:
But this it not related to the issue which I'm fixing. The problem was that combobox was not expanded while `assert_and_click 'yast2-lang_detailed-locale-setting';` was performed.

I've changed only the expanding procedure, I did not change any other actions like selecting "en_GB" in the dropdown list on the next step, where your suggestion may be applied. Sure, we can improve this but in scope of another task, as this is not related to the issue.

I guess the confusing part is `send_key_until_needlematch('yast2-lang_detailed-locale-setting_en_GB', 'spc', 5, 2);`. I'll explain in details: with `alt-d` I'm making the dropdown to be in focus, then with this `send_key_until_needlematch` I'm pressing `space` and checking if the dropdown is expanded. To avoid creating more needles, I've used the exising one - `yast2-lang_detailed-locale-setting_en_GB`, as the en_GB option is shown in the expanded list, so that means that the list is expanded. Then the test goes with the behavior as it was before.

Reply:
The issue is "language not changed" as it stated in the ticket and the root cause of the problem is the api function that does not work well on that arch,  how we solve it is up to us. When I was trying on my laptop I selected the combobox (which we can assume occurs in prod) and then just clicking up several time I got to the item (and we can assume we already have the needle for that) so that is more simple than send magic 'spc' key that is not really meaningful at least you put some comment on the code. Wondering why this code is not more simple? which avoid the api function and is clicking up on a combo to select a value...?
```
assert_screen 'yast2-lang_detailed-locale-setting';
send_key 'alt-d';
send_key_until_needlematch('yast2-lang_detailed-locale-setting_en_GB', 'up');
assert_screen 'yast2-lang_detailed-locale-setting_en_GB';
```
Anyway, if it is a matter of taste I can merge it, I was curious of your opinion and how you compare this two solutions. Maybe I'm wrong and I learn something from this :) or is just taste for code.